### PR TITLE
Introduce MSI-based installer

### PR DESCRIPTION
**Pull request by Rpinski**
Relates to: https://github.com/icsharpcode/ILSpy/issues/2490

### Problem
We would like to switch to a MSI-based standard installer, since it's now supported by the new Windows Store.

### Solution
* Set up a builder project with WixSharp to generate an MSI package

### CODE REVIEW

File: ILSpy.WithMsiBuilder.sln [hunk 1]
```diff
@@ -0,0 +1,135 @@
+﻿
```
Comment (siegfriedpammer) [inline ILSpy.WithMsiBuilder.sln:1]: Can we add this to `ILSpy.sln` and create an `ILSpy.Installer.slnf` (solution filter)?

File: ILSpy.sln [hunk 2]
```diff
@@ -47,6 +47,13 @@ Project("{D954291E-2A0B-460D-934E-DC6B0785DB48}") = "ILSpy.AddIn.Shared", "ILSpy
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "ILSpy.AddIn.VS2022", "ILSpy.AddIn.VS2022\ILSpy.AddIn.VS2022.csproj", "{09A03980-D14A-4705-A38C-741AD7166DEE}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ILSpy.Installer", "ILSpy.Installer\ILSpy.Installer.csproj", "{A4BA0771-DA4A-4A94-A5EC-5BA10B52816F}"
```
Comment (siegfriedpammer) [inline ILSpy.sln:50]: OK, then please just create an additional `ILSpy.Installer.slnf`. Thanks!

**Review by siegfriedpammer — CHANGES_REQUESTED** [review]

File: ILSpy.WithMsiBuilder.sln (hunk 1 above)
Comment (Rpinski) [reply ILSpy.WithMsiBuilder.sln:1]: Yep, just forgot to remove the file 🙈

File: .gitignore [hunk 3]
```diff
@@ -1,5 +1,6 @@
 bin/
 obj/
+packages/
```
Comment (siegfriedpammer) [inline .gitignore:3]: This change shouldn't be necessary anymore... package references are cached centrally per machine.

File: ILSpy.WithMsiBuilder.sln (hunk 1 above)
Comment (Rpinski) [reply ILSpy.WithMsiBuilder.sln:1]: Done

File: .gitignore (hunk 3 above)
Comment (Rpinski) [reply .gitignore]: Done

**Comment by christophwille** [comment]
@dgrunwald The idea is that most of the time we work with the .wpf slnf or alternatively unload the installer project in the .sln to not build the installer each and every time.

**Comment by siegfriedpammer** [comment]
Is this ready for review?

**Review by siegfriedpammer — APPROVED** [review]

**Pull request merged**